packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -97,31 +102,78 @@ class InMemoryAnchorService extends AnchorService {
     }
 
     for (const compositeKey of Object.keys(validCandidates)) {
-      const candidates: Candidate[] = validCandidates[compositeKey]
-
-      let nonce = 0
-      let selected: Candidate = null
-
-      for (const candidate of candidates) {
-        const record = (await this._ceramic.ipfs.dag.get(candidate.cid)).value
-
-        let currentNonce
-        if (DoctypeUtils.isSignedRecord(record)) {
-          const payload = (await this._ceramic.ipfs.dag.get(record.link)).value
-          currentNonce = payload.header?.nonce || 0
-        } else {
-          currentNonce = record.header?.nonce || 0
+      let candidates: Candidate[] = validCandidates[compositeKey]
+      candidates = await Promise.all(candidates.map(async c => {
+        c.partialLog.push(...await this._loadCommitHistory(c.cid))
+        return c
+      }))
+      candidates.sort((c1, c2) => c2.timestamp - c1.timestamp)
+
+      // naive implementation for finding the "valid" commit for the document.
+      // the decision making is going to change once the anchoring service becomes aware of Doctypes
+      let selected: Candidate
+      for (const c1 of candidates) {
+        selected = c1
+        let isIncluded = false
+
+        for (const c2 of candidates) {
+          if (c1 === c2) {
+            continue
+          }
+
+          if (c2.partialLog.includes(c1.cid)) {
+            isIncluded = true
+            break
+          }
         }
-        if (selected == null || currentNonce > nonce) {
-          selected = candidate
-          nonce = currentNonce
+
+        if (!isIncluded) {
+          result.push(selected)
+          break
         }
       }
-      result.push(selected)
     }
     return result
   }
 
+  /**
+   * Load last "depth" commits
+   *
+   * Note: this method will be replaced once CAS becomes aware of documents, not just individual commits
+   *
+   * @param commitId - Start CID
+   * @param depth - Number of history commits
+   * @private
+   */
+  async _loadCommitHistory(commitId: CID, depth = 10): Promise<CID[]> {
```

Comment:
why 10?  Seems like we should load the entire history, not stop at some arbitrary point.

Reply:
👍